src/main/java/org/opentripplanner/routing/algorithm/mapping/RaptorPathToItineraryMapper.java

```diff
@@ -114,24 +113,23 @@ public Itinerary createItinerary(Path<TripSchedule> path) {
         return itinerary;
     }
 
-    private void mapAccessLeg(
-            List<Leg> legs,
-            AccessPathLeg<TripSchedule> accessPathLeg
-    ) {
+    private List<Leg> mapAccessLeg(AccessPathLeg<TripSchedule> accessPathLeg) {
```

Comment:
Code cleanup: A basic principle of programming is to have methods with no side-effects. So, I changed this method to return the mapped legs, not map and insert into the result.